File: frontend/obfuscate_env.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function obfuscate(value) {
  if (!value) return '';
  const buffer = Buffer.from(value, 'utf-8');
  return 'obf:' + buffer.toString('base64');
}

function obfuscateDotenv(filepath) {
  if (!fs.existsSync(filepath)) {
    console.error(`Error: ${filepath} does not exist.`);
    return;
  }
  
  const content = fs.readFileSync(filepath, 'utf-8');
  const lines = content.split(/\r?\n/);
  const updatedLines = [];
  let updated = false;
  
  for (const line of lines) {
    const trimmed = line.trim();
    if (trimmed && !trimmed.startsWith('#') && trimmed.includes('=')) {
      const parts = line.split('=');
      const key = parts[0].trim();
      let val = parts.slice(1).join('=').trim();
      
      if (val && !val.startsWith('obf:')) {
        for (const quote of ['"', "'"]) {
          if (val.startsWith(quote) && val.endsWith(quote) && val.length >= 2) {
            val = val.slice(1, -1);
            break;
          }
        }
        const obfVal = obfuscate(val);
        updatedLines.push(`${key}=${obfVal}`);
        console.log(`Obfuscated ${key}`);
        updated = true;
      } else {
        updatedLines.push(line);
      }
    } else {
      updatedLines.push(line);
    }
  }
  
  if (updated) {
    fs.writeFileSync(filepath, updatedLines.join('\n'), 'utf-8');
    console.log(`Successfully obfuscated variables in ${filepath}`);
  } else {
    console.log(`No plain-text variables to obfuscate in ${filepath}`);
  }
}

const envPath = path.resolve(__dirname, '.env');
console.log(`Scanning ${envPath} for variables to obfuscate...`);
obfuscateDotenv(envPath);

File: backend/obfuscate_env.py
import os
import sys

# Add the current directory to python path so we can import app
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from app.core.security import obfuscate

def obfuscate_dotenv(filepath: str):
    if not os.path.exists(filepath):
        print(f"Error: {filepath} does not exist.")
        return
        
    lines = []
    updated = False
    with open(filepath, 'r', encoding='utf-8') as f:
        for line in f:
            stripped = line.strip()
            if stripped and not stripped.startswith('#') and '=' in line:
                parts = line.split('=', 1)
                key = parts[0].strip()
                val = parts[1].strip()
                # If there's a comment at the end of the line, like: VAL # comment
                # we should be careful. But standard .env usually has simple key=val or quotes.
                if val and not val.startswith('obf:'):
                    # Remove surrounding quotes if any
                    for quote in ['"', "'"]:
                        if val.startswith(quote) and val.endswith(quote) and len(val) >= 2:
                            val = val[1:-1]
                            break
                    obf_val = obfuscate(val)
                    lines.append(f"{key}={obf_val}\n")
                    print(f"Obfuscated {key}")
                    updated = True
                else:
                    lines.append(line)
            else:
                lines.append(line)
                
    if updated:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.writelines(lines)
        print(f"Successfully obfuscated variables in {filepath}")
    else:
        print(f"No plain-text variables to obfuscate in {filepath}")

if __name__ == '__main__':
    base_dir = os.path.dirname(os.path.abspath(__file__))
    env_path = os.path.join(base_dir, ".env")
    print(f"Scanning {env_path} for variables to obfuscate...")
    obfuscate_dotenv(env_path)
